pkg/reconciler/pingsource/jobrunner/pingsource.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobrunner
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/robfig/cron"
+	"knative.dev/eventing/pkg/apis/eventing"
+	"knative.dev/eventing/pkg/apis/sources/v1alpha2"
+
+	"go.uber.org/zap"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	sourceslisters "knative.dev/eventing/pkg/client/listers/sources/v1alpha2"
+
+	"knative.dev/eventing/pkg/logging"
+	"knative.dev/eventing/pkg/reconciler"
+	"knative.dev/pkg/controller"
+)
+
+// Reconciler reconciles PingSources
+type Reconciler struct {
+	*reconciler.Base
+
+	cronRunner       *cronJobsRunner
+	pingsourceLister sourceslisters.PingSourceLister
+	impl             *controller.Impl
```

Comment:
This looks unused, and is kind of a strange thing to have the reconciler have as a field (it creates a cycle in the layering).

Reply:
it's coming from the in-mem reconciler. Same problem there...